modal/image.py

```diff
@@ -120,6 +127,36 @@ def _make_pip_install_args(
     return args
 
 
+def _get_image_builder_version(client_version: ImageBuilderVersion) -> ImageBuilderVersion:
+    if config_version := config.get("image_builder_version"):
+        version = config_version
+        if (env_var := "MODAL_IMAGE_BUILDER_VERSION") in os.environ:
+            version_source = f" (based on your `{env_var}` environment variable)"
+        else:
+            version_source = f" (based on your local config file at `{user_config_path}`)"
+    else:
+        version = client_version
+        version_source = ""
+
+    supported_versions: Set[ImageBuilderVersion] = set(get_args(ImageBuilderVersion))
+    if version not in supported_versions:
+        if config_version is not None:
+            update_suggestion = "or remove your local configuration"
+        elif version < min(supported_versions):
+            update_suggestion = "your image builder version using the Modal dashboard"
+        else:
+            update_suggestion = "your client library (pip install --upgrade modal)"
```

Comment:
Does this mean the version is so old, we don't even have a reference to it in our client library? If yes, the user would have to update the library and also update the builder version somewhere right?

Reply:
Yeah, that's right — I think there are scenarios where you'd update the version in your dashboard and break your client, e.g. for a shared workspace, or if you have a computer you don't use very often.

Reply:
Btw, the question of what you do if your *workspace* version becomes entirely unsupported is something we haven’t solved yet, but probably wouldn’t be primarily handled in the client. That said, I’m imagining that we could also send a list of currently supported server versions back in ClientHello and use that here when the time comes.